crates/ruff_python_formatter/src/other/match_case.rs

```diff
@@ -21,24 +20,7 @@ impl FormatNodeRule<MatchCase> for FormatMatchCase {
         let comments = f.context().comments().clone();
         let dangling_item_comments = comments.dangling_comments(item);
 
-        write!(
-            f,
-            [
-                text("case"),
-                space(),
-                format_with(|f: &mut PyFormatter| {
-                    let comments = f.context().comments();
-
-                    for comment in comments.leading_trailing_comments(pattern) {
-                        // This is a lie, but let's go with it.
-                        comment.mark_formatted();
-                    }
-
-                    // Replace the whole `format_with` with `pattern.format()` once pattern formatting is implemented.
-                    not_yet_implemented_custom_text("NOT_YET_IMPLEMENTED_Pattern", pattern).fmt(f)
-                }),
-            ]
-        )?;
+        write!(f, [text("case"), space(), pattern.format()])?;
```

Comment:
This now fails for cases where the pattern has comments. We may need to preserve parentheses similar to the `WithItem` formatting.

Reply:
@MichaReiser This is fixed